From 7646269ccf4df42f15d6fea4d9b4f1e9281d3b02 Mon Sep 17 00:00:00 2001
From: Mingyuan Qi <mingyuan.qi@intel.com>
Date: Thu, 11 Apr 2019 14:59:11 +0800
Subject: [PATCH 11/11] Ironic: Add pxe boot support for centos image

Current script does not consider centos distro as base image.
Different folder was checked to copy pxe files to tftpboot folder.

Signed-off-by: Mingyuan Qi <mingyuan.qi@intel.com>
---
 .../bin/_ironic-conductor-pxe-init.sh.tpl          | 23 +++++++++++++++++-----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/ironic/templates/bin/_ironic-conductor-pxe-init.sh.tpl b/ironic/templates/bin/_ironic-conductor-pxe-init.sh.tpl
index b8c4c4c..164874f 100644
--- a/ironic/templates/bin/_ironic-conductor-pxe-init.sh.tpl
+++ b/ironic/templates/bin/_ironic-conductor-pxe-init.sh.tpl
@@ -18,17 +18,30 @@ limitations under the License.
 
 set -ex
 
-#NOTE(portdirect): this works round a limitation in Kolla images
-if ! dpkg -l ipxe; then
-  apt-get update
-  apt-get install ipxe -y
+DISTRO_UBUNTU=$(cat /etc/*release | grep Ubuntu)
+
+if [[ $DISTRO_UBUNTU != "" ]]; then
+  #NOTE(portdirect): this works round a limitation in Kolla images
+  if ! dpkg -l ipxe; then
+    apt-get update
+    apt-get install ipxe -y
+  fi
 fi
 
 mkdir -p /var/lib/openstack-helm/tftpboot
 mkdir -p /var/lib/openstack-helm/tftpboot/master_images
 
-for FILE in undionly.kpxe ipxe.efi; do
+for FILE in undionly.kpxe ipxe.efi pxelinux.0; do
   if [ -f /usr/lib/ipxe/$FILE ]; then
     cp -v /usr/lib/ipxe/$FILE /var/lib/openstack-helm/tftpboot
   fi
+
+  # For CentOS
+  if [ -f /var/lib/tftpboot/$FILE ]; then
+    cp -v /var/lib/tftpboot/$FILE /var/lib/openstack-helm/tftpboot
+  fi
+  if [ -f /usr/share/ipxe/$FILE ]; then
+    cp -v /usr/share/ipxe/$FILE /var/lib/openstack-helm/tftpboot
+  fi
+
 done
-- 
1.8.3.1

